File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/ILocalPhotoAlbumsView.java
package dev.ragnarok.fenrir.mvp.view;

import androidx.annotation.NonNull;

import java.util.List;

import dev.ragnarok.fenrir.model.LocalImageAlbum;
import dev.ragnarok.fenrir.mvp.core.IMvpView;


public interface ILocalPhotoAlbumsView extends IMvpView {

    void displayData(@NonNull List<LocalImageAlbum> data);

    void setEmptyTextVisible(boolean visible);

    void displayProgress(boolean loading);

    void openAlbum(@NonNull LocalImageAlbum album);

    void notifyDataChanged();

    void requestReadExternalStoragePermission();
}

File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/ILocalPhotoAlbumsView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.LocalImageAlbum
import dev.ragnarok.fenrir.mvp.core.IMvpView

interface ILocalPhotoAlbumsView : IMvpView {
    fun displayData(data: List<LocalImageAlbum>)
    fun setEmptyTextVisible(visible: Boolean)
    fun displayProgress(loading: Boolean)
    fun openAlbum(album: LocalImageAlbum)
    fun notifyDataChanged()
    fun requestReadExternalStoragePermission()
}